core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2Metadata.java

```diff
@@ -130,7 +147,64 @@ public String toString() {
      */
     @Override
     public String get(String path) {
-        return "IMDS";
+
+        String data = null;
+        AbortableInputStream abortableInputStream = null;
+        try {
+            String token = getToken();
+            URI uri = URI.create(endpoint + path);
```

Comment:
Is there room for error if the endpoint has a trailing slash in it that we might have endpoint//path?

Reply:
I am implementing that as part of endpoint resolution feature.